base/channels.jl

```diff
@@ -53,13 +54,82 @@ end
 
 Channel(sz) = Channel{Any}(sz)
 
+# special constructors
+"""
+    Channel(func::Function; ctype=Any, csize=0, taskref=nothing)
+
+Creates a new task from `func`, binds it to a new channel of type
+`ctype` and size `csize`, schedules the task, all in a single call.
+
+`func` must accept the bound channel as its only argument.
+
+If you need a reference to the created task, pass a `Ref{Task}` object via
+keyword argument `taskref`.
+
+Returns a Channel.
+
+```jldoctest
+julia> chnl = Channel(c->foreach(i->put!(c,i), 1:4));
+
+julia> @show typeof(chnl);
+typeof(chnl) = Channel{Any}
+
+julia> for i in chnl
+           @show i
+       end;
+i = 1
+i = 2
+i = 3
+i = 4
+
+```
+
+An example of referencing the created task:
+
+```jldoctest
+julia> taskref = Ref{Task}();
+
+julia> chnl = Channel(c->(@show take!(c)); taskref=taskref);
+
+julia> task = taskref[];
```

Comment:
Yes, but you typically need to ensure that the Task is bound before it is scheduled to handle early errors in the task. For example,
```
c = Channel()
t = @schedule (foo(); while(some_condition); put!(c,data); end)
bind(c,t)
for x in c
  bar(x)
end 
```

can lead to the for-loop missing any exception thrown by `foo`. Actually, it is currently not an issue since `@chedule` does not yield, but it is safe to assume that this behavior can change in the future.   

A safer way is
```
c = Channel()
t = Task(()->(foo(); while(some_condition); put!(c,data); end))
bind(c,t)
schedule(c)
for x in c
  bar(x)
end 
```


With the special Channel constructor, the above code block is equivalent to
```
for x in Channel(c->(foo(); while(some_condition); put!(c,data); end))
  bar(x)
end
```